PlutoHelperAgent/AppDelegate.m

```diff
@@ -231,12 +233,31 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
         
     
         [ProjectLockerAndKeychainFunctions check_logged_in:^(enum ReturnValues connectionStatus) {
-            
+
             switch(connectionStatus){
                 case ALLOK:
                     [self tryOpenProject:projectid];
                     break;
                 case PERMISSION_DENIED:
+                    if (connectionAttempts == 0) {
```

Comment:
I'll accept this for the time being but the code seems to be getting a bit spaghettified, will have a think how we could maybe refactor this into something easier to read.